src/proc.cpp

```diff
@@ -794,12 +791,16 @@ bool terminal_give_to_job(const job_t *j, bool cont) {
         // 4.4.0), EPERM does indeed disappear on retry. The important thing is that we can
         // guarantee the process isn't going to exit while we wait (which would cause us to possibly
         // block indefinitely).
-        while (tcsetpgrp(STDIN_FILENO, j->pgid) != 0) {
-            debug(3, "tcsetpgrp failed");
+        while (true) {
```

Comment:
I actually prefer the old loop condition.

Reply:
I agree. I think during refactoring and rebasing a billion times over the final code works by simply plugging `tcsetpgrp` back into the condition but at some point in the middle that wasn't possible. Reverted.